From 556e2830a1de15359c261fe854ddb51a6a541777 Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexdeucher@gmail.com>
Date: Tue, 9 Feb 2010 18:22:00 -0500
Patch-mainline: 2.6.34
References: fate#310916
Git-commit: 4c36b678a23d33b82d614afe4f958a9d244ede5d
Subject: [PATCH 0419/2588] drm/radeon/kms/evergreen: adapt to i2c changes

original evergreen patches we against a kernel tree
without my radeon i2c algo changes.

Signed-off-by: Alex Deucher <alexdeucher@gmail.com>
Signed-off-by: Dave Airlie <airlied@redhat.com>
(cherry picked from commit 4c36b678a23d33b82d614afe4f958a9d244ede5d)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/radeon/radeon_i2c.c |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/radeon/radeon_i2c.c b/drivers/gpu/drm/radeon/radeon_i2c.c
index d15fa5e..4ae50c1 100644
--- a/drivers/gpu/drm/radeon/radeon_i2c.c
+++ b/drivers/gpu/drm/radeon/radeon_i2c.c
@@ -254,6 +254,13 @@ static u32 radeon_get_i2c_prescale(struct radeon_device *rdev)
 	case CHIP_RV740:
 		/* todo */
 		break;
+	case CHIP_CEDAR:
+	case CHIP_REDWOOD:
+	case CHIP_JUNIPER:
+	case CHIP_CYPRESS:
+	case CHIP_HEMLOCK:
+		/* todo */
+		break;
 	default:
 		DRM_ERROR("i2c: unhandled radeon chip\n");
 		break;
@@ -833,6 +840,14 @@ static int radeon_i2c_xfer(struct i2c_adapter *i2c_adap,
 		/* XXX fill in hw i2c implementation */
 		ret = radeon_sw_i2c_xfer(i2c_adap, msgs, num);
 		break;
+	case CHIP_CEDAR:
+	case CHIP_REDWOOD:
+	case CHIP_JUNIPER:
+	case CHIP_CYPRESS:
+	case CHIP_HEMLOCK:
+		/* XXX fill in hw i2c implementation */
+		ret = radeon_sw_i2c_xfer(i2c_adap, msgs, num);
+		break;
 	default:
 		DRM_ERROR("i2c: unhandled radeon chip\n");
 		ret = -EIO;
-- 
1.7.6

